src/test/java/org/elasticsearch/test/rest/ElasticsearchRestTestCase.java

```diff
@@ -302,4 +321,86 @@ public void test() throws IOException {
             executableSection.execute(restTestExecutionContext);
         }
     }
+
+    // don't look any further: NO TOUCHY!
+    
+    public static class Rest0Tests extends ElasticsearchRestTestCase {
```

Comment:
It is more than that. The original logic (commented out) was not reproducible. Thats because at the minimum, it divided up its tests based on *full path* which changes depending on the machine its run on, can even change every test execution on the same machine.

Doing sneaky stuff like this should really be avoided. I hated doing it but the rest tests leave the tests so unbalanced that something had to be done. I took time to make sure everything was reproducible, it might be ugly, but it works. 

Making a test fast, but no longer reproducible is no good. In that case we should just slap `@Ignore` on it.